mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -438,9 +466,36 @@ LogicalResult SYCLNDRangeConstructorOp::verify() {
   return emitBadSignatureError(*this);
 }
 
-void SYCLNDRangeConstructorOp::getCanonicalizationPatterns(
-    RewritePatternSet &patterns, MLIRContext *context) {
-  patterns.add<SYCLNDRangeConstructorEraseDefaultOffset>(context);
+OpFoldResult SYCLNDRangeConstructorOp::fold(FoldAdaptor) {
+  bool mutated = false;
+
+  /// Transform:
+  /// ```
+  /// %off = sycl.id.constructor() : () -> memref<?x!sycl_id_X_>
+  /// sycl.nd_range.constructor(%gs, %ls, %off)
+  ///   : (memref<?x!sycl_range_X_>,
+  ///      memref<?x!sycl_range_X_>,
+  ///      memref<?x!sycl_id_X_>)
+  ///   -> memref<?x!sycl_nd_range_X_>
+  /// ```
+  /// To:
+  /// ```
+  /// sycl.nd_range.constructor(%gs, %ls)
+  ///   : (memref<?x!sycl_range_X_>, memref<?x!sycl_range_X_>)
+  ///   -> memref<?x!sycl_nd_range_X_>
+  /// ```
+  constexpr unsigned offsetOperandIndex = 2;
+  MutableOperandRange args = getArgsMutable();
+  if (args.size() == offsetOperandIndex + 1 &&
+      isDefaultConstructedID(args[offsetOperandIndex].get())) {
+    args.erase(offsetOperandIndex);
+    mutated = true;
```

Comment:
Is it idiomatic in MLIR to modify operations in place rather than replace them with a new operation?